contribute.md

```diff
@@ -3,104 +3,345 @@ layout: contribute
-
 ## Contributing Code
 
-### Read the Foreman [handbook](/handbook.html)
-The Foreman is an open-source project that's licensed under the GNU General Public License version 3.<br>
-Contributions of all types are gladly accepted!
+The Foreman is an open-source project that's licensed under the GNU General Public License version 3. Contributions of all types are gladly accepted! Before setting up your development environment, check out our [issue tracker](https://projects.theforeman.org/projects). We have a list of [trivial or easy](https://projects.theforeman.org/projects/foreman/issues?query_id=42) issues in the Foreman core project. Similar queries can be done in other projects on the issue tracker.
+
+Before writing any code, be sure to read the Foreman [handbook](/handbook.html) for some code guidelines and recommendations. You can find us on both our forum and IRC if you run into troubles when picking issues or setting up your development environment. See our [Support](/support) page for all the details on how to come chat with us.
 
 ### Setting up a development environment
-These types of tasks generally require a familiarity with Ruby (on Rails) development or RPM/Debian packaging. If you are still new to Rails, community members can help you if you get stuck with something or have any other questions.
+
+These types of tasks generally require a familiarity with Linux, Ruby, Ruby on Rails, Javascript and React development. If you are still new to Rails, community members can help you if you get stuck with something or have any other questions.
+
+To get you started, we've prepared a short video demonstrating the process of both documented options.
+
+TODO: Put the youtube lzaps video here.
+
+#### Option A: Preconfigured virtual machine
 
 You will need to download a copy of the current development code. [The
 official code repository](https://github.com/theforeman) is located
 on GitHub. All changes are made in the `develop` branch.
 
-There is a tool called [Forklift](https://github.com/theforeman/forklift) that can be used to automate deploying Foreman development environments.
+There is a tool called [Forklift](https://github.com/theforeman/forklift) that can be used to automate deploying Foreman development environments. This will work on Linux, MacOS or even Windows and the only required tools are Vagrant (virtual machine manager and image deployment tool), Ansible (configuration management tool) and a host with virtualization capabilities.
+
+There are couple of installation profiles available, namely:
+
+* centos7-katello-devel: Git clones of foreman, remote execution and katello plugins and all required backend systems running on CentOS 7.
+* centos7-luna-devel: Git clones of foreman and all plugins which are shipped as Red Hat Satellite product running on CentOS 7.
+* centos7-foreman-nightly: Installation of nightly build of Foreman from RPM packages running on CentOS 7. This can be used for evaluation or pilot installation.
```

Comment:
These are good examples, but I think we should mention that besides CentOS Linux 7 there are also other distributions and versions. Also, it was requested by the CentOS folks to start using CentOS Linux to differentiate with CentOS Stream.
```suggestion
* centos7-foreman-nightly: Installation of nightly build of Foreman from RPM packages running on CentOS Linux 7. This can be used for evaluation or pilot installation. You can substitute `centos7` with `centos8`, `centos8-stream`, `debian10` or `ubuntu1804` to get another base OS. Similarly, `nightly` can be replaced by `2.4` or any other supported version. Instead of `foreman`, `katello` can also be used.
```

That's probably too long though.